Added AI usage guideline for the application process

adamblanchard opened the pull request:
We need to share clear(er) expectations with applicants to HYF about our expectations around the usage of AI.

This is the first version of that document. It will be shared with applicants during the application process.

I imagine we iterate these expectations quite often, as we learn more about how new teams find working with this wording, and also as AI develops in the industry.

In general, the staff team have agreed to this general usage of AI, so I'm requesting feedback mostly on the wording, examples, and how to make this as clear as possible in writing.

However, if you have differing opinions on how we should/shouldn't allow the usage of AI, feel free to share here.

te-online commented on courses/Pre-Course/guidelines/ai-usage.md:
```diff
@@ -0,0 +1,34 @@
+
+While the world is exploding with the introduction of AI, and changing rapidly every day, it’s unclear still on the long term impact and changes AI will have in our day to day work and lives.
+
+For now, we know AI is a useful tool in many areas - we also use it internally at HYF! Therefore, to maintain realism and fairness, we aim to support a reasonable and effective usage of AI. During your time at HYF, you will also have the opportunity to learn more about the practical, effective and ethical use of AI in the workplace.
```
> During your time at HYF, you will also have the opportunity to learn more about the practical, effective and ethical use of AI in the workplace.

I'd be hesitant to express it like that. The "practical, effective and ethical" (especially "ethical") use of AI is pretty much a work-in-progress, to say it with euphemism. Maybe we have mentors that can give good input on the use of Generative AI, but I'd be careful to promise too much here.

adamblanchard replied:
I understand where you're coming from.

We are planning to build/introduce AI at least during one session. But i think until we have that outlined and actually in the curriculum it's not safe to promise what will be included there.

For now i will reduce this to "practical use" only. 

Thanks!

te-online commented:
First of all, great to see guidelines on the elephant in the room, Generative AI! ❤️ 

I think the approach you outline is quite clear and easy to understand. And it seems very much in line with what I have seen in other education-related guidelines (though I have not been seeking out many of those!).

There's one thing that I would imagine to help tremendously in understanding the Dos and Dont's. **Giving example prompts.** This is what [OpenAI's own guide uses](https://openai.com/chatgpt/use-cases/student-writing-guide/), in my opinion quite successfully.

This document may not be the place for it, but I would really like to argue and advertise for including a section about **AI literacy** (not necessarily calling it that, to avoid "the old teacher tone of voice"). 

- We may want to consider at least introducing the term "Generative AI" once (then we can call it AI going forward), since it seems like the current most appropriate description of ChatGTP-like tools. ("Traditional" autocomplete in my IDE is also AI, but not generative). This may seem insignificant, but with a term changing so rapidly, I think it could pay off to be as precise as can be at the moment of writing – and updating it regularly to match common understanding of the term
- We may want to add the fact that ChatGTP cannot replace reading trustworthy and peer-reviewed resources, like MDN. 
- It is also important to stress, that it is **not a search engine**, even though most people use it like that. The main reason being hallucinations
  1. Content generated by Generative AI is more likely to be incorrect than content from trustworthy, established, and possibly peer-reviewed sources, due to the nature of how language models work 
  2. When Generative AI is telling a lie, it's not at all obvious. A bad answer will look like a good answer, but can be totally wrong. There are no telltale-signs of a bad answer (like bad grammar, formatting etc.)
- We may want to advertise for utilizing the fact that applicants can run code: Don't trust the Generative AI on an explanation – try it out instead. Does it work? See for yourself

---

I've left two additional comments in context.

--- 

Thanks for the good work and great effort, staff team! ☺️

adamblanchard commented:
Just a couple of personal notes to maybe include:
- A reasonable way to allow "feedback on communication" (without it just writing it all for you)
- Maybe something like "while some devs often use AI to generate code solutions, we're focussing on building your foundational skills/knowledge during hyf..."

adamblanchard commented on courses/Pre-Course/guidelines/ai-usage.md:
```diff
@@ -0,0 +1,34 @@
+
+## HackYourFuture’s stance on AI
+
+While the world is exploding with the introduction of AI, and changing rapidly every day, it’s unclear still on the long term impact and changes AI will have in our day to day work and lives.
```
done

adamblanchard commented:
I've updated this guideline to not only cover the applcation process, but also a guideline for ai usage for trainees during the programme, because i think it's basically the same.

@te-online I resolved your suggestions in the code, and added some basic prompt examples for now. I've added your suggestions for literacy stuff into the module proposal [here](https://github.com/HackYourFuture-CPH/programme/pull/126/files#diff-37cac4f8bc8546e45596a7cfacb23cf577585ef60e3681b91ba87a3fcb97e58a) so they won't be lost.

I also reviewd and updated some of the wordings and sentences since the last review to make them a little clearer.

jason-vasilev commented on guidelines/ai-usage.md:
```diff
@@ -0,0 +1,50 @@
+# Using Generative AI at HackYourFuture
+
+During your time at HackYourFuture, whether you're applying to join and in the application process, or already on the programme, you may wish to use Generative AI (AI) in some form to help you out on a task. Using AI includes using tools such as ChatGPT, GitHub Copilot and similar. This guideline aims to explain how HackYourFuture views AI usage, and our expectations for how you should (and shoul ...
```
Sounded a bit strange to me, and changed the order of the first sentence.
```suggestion
Whether you're applying to join HackYourFuture or are already enrolled in the programme, you may consider using Generative AI tools to solve various problems. This includes tools like ChatGPT, GitHub Copilot, and similar. This guideline outlines how HackYourFuture views the use of AI and our expectations for how you should (and should not) use it while working with us.
```

jason-vasilev commented on guidelines/ai-usage.md:
```diff
@@ -0,0 +1,50 @@
+During your time at HackYourFuture, whether you're applying to join and in the application process, or already on the programme, you may wish to use Generative AI (AI) in some form to help you out on a task. Using AI includes using tools such as ChatGPT, GitHub Copilot and similar. This guideline aims to explain how HackYourFuture views AI usage, and our expectations for how you should (and shoul ...
+
+> [!WARNING]
+> It’s important you follow these guidelines. If we find evidence of you misusing AI based on the below, you may be rejected from the application process or removed from the programme.
```
```suggestion
> It’s important that you follow these guidelines. If we find evidence that you’ve misused AI as outlined below, you may be rejected from the application process or removed from the programme."
```

jason-vasilev commented on guidelines/ai-usage.md:
```diff
@@ -0,0 +1,50 @@
+
+## How you can use AI
+
+Used effectively, AI can be a great tool to support your learning process and skill development. However, if you're not careful, it can hinder your progress and take away important opportunities to learn.
```
We've used "journey" a few times already across the programme
```suggestion
Used effectively, AI can be a great tool to support your learning journey and skill development. However, if you're not careful, it can hinder your progress and take away important learning opportunities.
```

jason-vasilev commented on guidelines/ai-usage.md:
```diff
@@ -0,0 +1,50 @@
+
+## How your experience and knowledge will be evaluated
+
+While you may choose to use AI (following the guidelines above), please be aware that **you will be required to be able to explain all of your answers and code** during your time at HackYourFuture. Make sure when you create, submit or share code or solutions, that you fully understand how it works based on your own knowledge.
```
Softened the language a bit by changing "required" to "expected".
Moved "make sure" to more appropriate place
```suggestion
While you're welcome to use AI tools (following the guidelines above), please be aware that **you'll be expected to clearly explain all your answers and code** during your time at HackYourFuture. Whenever you create, submit, or share code or solutions, make sure you fully understand how they work based on your own knowledge.
```

jason-vasilev reviewed (approved):
A bit too late, but here are a few suggestions on improving the language.

Else, I am also excited about the guidelines, and can contribute with this article from Canva on how they've changed their interview processes by allowing people to use AI in the technical tasks - https://www.canva.dev/blog/engineering/yes-you-can-use-ai-in-our-interviews/

adamblanchard commented:
Thanks @jason-vasilev for giving this a review, appreciate your suggestions. I've put them all in a separate PR https://github.com/HackYourFuture-CPH/programme/pull/127.

HosseinZahed commented:
Thank you all for all the effort you've put in. It looks absolutely fine to me. Let's get started with this and move the niddle forward. 🙌🏻
